src/main.rs

```diff
@@ -69,15 +114,39 @@ fn main() -> anyhow::Result<()> {
         let formatted_raw = format!("({}, {}, {})", avg.x, avg.y, avg.z).italic();
         println!("Average: {formatted} {formatted_raw}\n");
 
-        println!("Number of entries: {n}");
+        let formatted = format!("({} discarded)", m - n).italic();
+        println!("Number of entries: {m} {}", formatted);
         let formatted = format!("({:.6}º, {:.6}º, {:.3}m)", std_dev.x, std_dev.y, std_dev.z);
-        let formatted_m = format!(
-            "Horizontally: ~({:.2}m, {:.2}m)",
-            std_dev.x * EARTH_RADIUS,
-            std_dev.y * EARTH_RADIUS
-        )
-        .italic();
+        let formatted_m =
+            format!("Horizontally: ~({:.2}m, {:.2}m)", std_dev_m.x, std_dev_m.y).italic();
         println!("Standard deviation: {formatted} {formatted_m}");
+
+        let formatted = {
+            let mut formatted =
+                String::from_str("  Latitude (º)            Longitude (º)           Altitude(m)\n")
+                    .unwrap();
+            let mut val_x = division_val_x.iter(); // All these iterators have the same length (2 * cutoff * div ) from histogram
```

Comment:
If that is true, you can just zip them https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip